src/views/virtualmachines/details/tabs/diagnostic/VirtualMachineLogViewer/VirtualMachineLogViewer.tsx

```diff
@@ -15,31 +16,44 @@ import VirtualMachineBasicLogViewer from './VirtualMachineBasicLogViewer/Virtual
 
 const VirtualMachineLogViewer = ({ connect, vm }) => {
   const { t } = useKubevirtTranslation();
-  const { pods, vmi } = useVMIAndPodsForVM(vm?.metadata?.name, vm?.metadata?.namespace);
+  const { loaded, pods, vmi } = useVMIAndPodsForVM(vm?.metadata?.name, vm?.metadata?.namespace);
   const pod = getVMIPod(vmi, pods);
   const { featureEnabled: isClusterDisabledGuestSystemLogs } = useFeatures(
     DISABLED_GUEST_SYSTEM_LOGS_ACCESS,
   );
+
   const isDisabledAtVM =
     (getDevices(vm) as V1Devices & { logSerialConsole: boolean })?.logSerialConsole === false &&
     !isClusterDisabledGuestSystemLogs;
 
   const isNeededRestart = getChangedGuestSystemAccessLog(vm, vmi);
 
-  const { data } = useVirtualMachineLogData({ connect, pod });
+  const isPodLogContainerExist = Boolean(
+    pod?.spec?.containers?.find((container) => container?.name === 'guest-console-log'),
+  );
 
-  if (!isRunning(vm)) {
-    return <Bullseye>{t('Virtual machine is not running')}</Bullseye>;
+  const { data } = useVirtualMachineLogData({ connect: connect && isPodLogContainerExist, pod });
+
+  if (!loaded) {
+    return (
+      <Bullseye>
+        <Loading />
+      </Bullseye>
+    );
   }
 
-  if (isNeededRestart) {
-    return <Bullseye>{t('Guest system logs not ready. Restart required')}</Bullseye>;
+  if (!isRunning(vm)) {
+    return <Bullseye>{t('Virtual machine is not running')}</Bullseye>;
   }
 
   if (isClusterDisabledGuestSystemLogs) {
     return <Bullseye>{t('Guest system logs are disabled at cluster')}</Bullseye>;
   }
 
+  if (isNeededRestart || !isPodLogContainerExist) {
+    return <Bullseye>{t('Guest system access logs not ready. a restart required')}</Bullseye>;
```

Comment:
Why are you adding this string here? Just recently it was suggested to me to use "Restart required" instead of "a restart required". And the sentence should start with a big letter.. Maybe you've just accidentally copied here the older version of this string?

Reply:
Probably beause moved the IF check and it took the string before. anyway changed